### PR TITLE
Introduce SAML sections for 'Sync user attributes during Sign in' & 'Allow subdomains options'

### PR DESCRIPTION
**Pull request by chanioxaris**
Introduce two new sections on the SAML overview page regarding the below options
- Sync user attributes during Sign in
- Allow subdomains

### CODE REVIEW

File: docs/authentication/saml/overview.mdx [hunk 1]
```diff
@@ -113,6 +113,25 @@ Now that everything is set up, you need to activate the SAML Connection via the
...
 Everything should be set up now for you to try out authentication via SAML. Go to your application’s Sign In page and add your email in the input field. If it matches an active SAML Connection, you will be redirected to your IdP where you will sign in with your credentials.
 
+## Sync user attributes during Sign in
+During SAML SSO and after a user successfully authenticated, the IdP provides Clerk with the corresponding user data. After each successful Sign in, we are making sure to keep the user data up to date
+based on the response of the SAML provider. If this behavior is something you are not looking into, you can disable it by visiting the SAML Connection `Advanced` tab in [Clerk Dashboard](https://dashboard.clerk.com/) and disable the `Sync user attributes during sign-in` option.
+
+This setting by default is enabled!
+
+## Allow subdomains
+SAML SSO is working based on domains and an email address domain must match the exact domain a SAML Connection has been configured with, in order to be able to use it to authenticate.
```
Comment (S3Prototype) [inline docs/authentication/saml/overview.mdx:123]:
```suggestion
Authenticating via SAML SSO requires the user's email address domain to match the exact domain the SAML Connection has been configured with.
```

File: docs/authentication/saml/overview.mdx [hunk 2]
```diff
@@ -113,6 +113,25 @@ Now that everything is set up, you need to activate the SAML Connection via the
...
+Although, there are some cases where companies are using subdomains in order to differentiate between different departments, but they are still looking to use a single SAML Connection for every user of theirs.
+
+To support such use case you can enable the `Allow subdomains` option, which you can find in the SAML Connection `Advanced` tab in [Clerk Dashboard](https://dashboard.clerk.com/). For example, a user with the email address `john@sales.example.com`
+will be able to use a SAML Connection with the `example.com` domain in order to authenticate.
+
+This setting by default is disabled!
```
Comment (S3Prototype) [inline docs/authentication/saml/overview.mdx:129]:
```suggestion
This setting is disabled by default.
```

File: docs/authentication/saml/overview.mdx [hunk 3]
```diff
@@ -113,6 +113,25 @@ Now that everything is set up, you need to activate the SAML Connection via the
...
+
+<Callout type="info">
+  In order to enable the Allow subdomains option, your SAML Connection domain must be an eTLD+1.
```
Comment (S3Prototype) [inline docs/authentication/saml/overview.mdx:132]:
```suggestion
  To enable the Allow subdomains option, your SAML Connection domain must be an eTLD+1.
```

File: docs/authentication/saml/overview.mdx [hunk 4]
```diff
@@ -113,6 +113,25 @@ Now that everything is set up, you need to activate the SAML Connection via the
...
+During SAML SSO and after a user successfully authenticated, the IdP provides Clerk with the corresponding user data. After each successful Sign in, we are making sure to keep the user data up to date
+based on the response of the SAML provider. If this behavior is something you are not looking into, you can disable it by visiting the SAML Connection `Advanced` tab in [Clerk Dashboard](https://dashboard.clerk.com/) and disable the `Sync user attributes during sign-in` option.
+
+This setting by default is enabled!
+
+## Allow subdomains
+SAML SSO is working based on domains and an email address domain must match the exact domain a SAML Connection has been configured with, in order to be able to use it to authenticate.
+Although, there are some cases where companies are using subdomains in order to differentiate between different departments, but they are still looking to use a single SAML Connection for every user of theirs.
+
+To support such use case you can enable the `Allow subdomains` option, which you can find in the SAML Connection `Advanced` tab in [Clerk Dashboard](https://dashboard.clerk.com/). For example, a user with the email address `john@sales.example.com`
```
Comment (S3Prototype) [inline docs/authentication/saml/overview.mdx:126]:
```suggestion
To support SAML SSO with subdomains, you must enable the **Allow subdomains** option in the [Clerk Dashboard](https://dashboard.clerk.com/). You can find this setting in the SAML Connection **Advanced**.

For example, a user with the email address `john@sales.example.com`
```

File: docs/authentication/saml/overview.mdx [hunk 5]
```diff
@@ -113,6 +113,25 @@ Now that everything is set up, you need to activate the SAML Connection via the
...
 
+## Sync user attributes during Sign in
+During SAML SSO and after a user successfully authenticated, the IdP provides Clerk with the corresponding user data. After each successful Sign in, we are making sure to keep the user data up to date
+based on the response of the SAML provider. If this behavior is something you are not looking into, you can disable it by visiting the SAML Connection `Advanced` tab in [Clerk Dashboard](https://dashboard.clerk.com/) and disable the `Sync user attributes during sign-in` option.
+
+This setting by default is enabled!
+
+## Allow subdomains
+SAML SSO is working based on domains and an email address domain must match the exact domain a SAML Connection has been configured with, in order to be able to use it to authenticate.
+Although, there are some cases where companies are using subdomains in order to differentiate between different departments, but they are still looking to use a single SAML Connection for every user of theirs.
```
Comment (S3Prototype) [inline docs/authentication/saml/overview.mdx:124]:
```suggestion
```

Sentence no longer needed with my suggestion below

**Review by S3Prototype — APPROVED** [review]
Approved with suggested changes

File: docs/authentication/saml/overview.mdx [hunk 6]
```diff
@@ -113,6 +113,25 @@ Now that everything is set up, you need to activate the SAML Connection via the
 
 Everything should be set up now for you to try out authentication via SAML. Go to your application’s Sign In page and add your email in the input field. If it matches an active SAML Connection, you will be redirected to your IdP where you will sign in with your credentials.
 
+## Sync user attributes during Sign in
+During SAML SSO and after a user successfully authenticated, the IdP provides Clerk with the corresponding user data. After each successful Sign in, we are making sure to keep the user data up to date
```
[the remark this answers is not in the document]
Comment (chanioxaris) [reply docs/authentication/saml/overview.mdx]: Updated!

File: docs/authentication/saml/overview.mdx [hunk 7]
```diff
@@ -113,6 +113,25 @@ Now that everything is set up, you need to activate the SAML Connection via the
...
+based on the response of the SAML provider. If this behavior is something you are not looking into, you can disable it by visiting the SAML Connection `Advanced` tab in [Clerk Dashboard](https://dashboard.clerk.com/) and disable the `Sync user attributes during sign-in` option.
```
[the remark this answers is not in the document]
Comment (chanioxaris) [reply docs/authentication/saml/overview.mdx]: Updated!

File: docs/authentication/saml/overview.mdx [hunk 8]
```diff
@@ -113,6 +113,25 @@ Now that everything is set up, you need to activate the SAML Connection via the
...
+
+This setting by default is enabled!
```
[the remark this answers is not in the document]
Comment (chanioxaris) [reply docs/authentication/saml/overview.mdx]: Updated!

File: docs/authentication/saml/overview.mdx (hunk 1 above)
Comment (chanioxaris) [reply docs/authentication/saml/overview.mdx]: Updated!

File: docs/authentication/saml/overview.mdx (hunk 2 above)
Comment (chanioxaris) [reply docs/authentication/saml/overview.mdx]: Updated!

File: docs/authentication/saml/overview.mdx (hunk 3 above)
Comment (chanioxaris) [reply docs/authentication/saml/overview.mdx]: Updated!

File: docs/authentication/saml/overview.mdx (hunk 4 above)
Comment (chanioxaris) [reply docs/authentication/saml/overview.mdx]: Updated!

File: docs/authentication/saml/overview.mdx (hunk 5 above)
Comment (chanioxaris) [reply docs/authentication/saml/overview.mdx]: Removed!

**Pull request merged**